### PR TITLE
Make http.url field opt-in

### PR DESCRIPTION
**Pull request by tl-rodrigo-gryzinski**
Per the discussion in #64, make `http.url` tracing opt-in and remove credentials from the URL before exposing it. Added a new span backend that does the same as the default one with the added URL field

### CODE REVIEW

File: reqwest-tracing/src/reqwest_otel_span_builder.rs [hunk 1]
```diff
@@ -112,6 +113,26 @@ fn get_header_value(key: &str, headers: &HeaderMap) -> String {
     format!("{:?}", headers.get(key).unwrap_or(header_default)).replace('"', "")
 }
 
+/// Similar to [`DefaultSpanBackend`] but also adds the `http.url` attribute to request spans.
+///
+/// [`TracingMiddleware`]: crate::middleware::TracingMiddleware
+pub struct SpanBackendWithUrl;
```
Comment (tl-rodrigo-gryzinski) [inline reqwest-tracing/src/reqwest_otel_span_builder.rs:119]: Not too happy with the name

**Review by conradludgate — DISMISSED** [review]
LGTM

File: reqwest-tracing/src/reqwest_otel_span_builder.rs (hunk 1 above)
Comment (conradludgate) [reply reqwest-tracing/src/reqwest_otel_span_builder.rs:119]: I was thinking we could have `WithUrl<DefaultSpanBackend>` but the tracing macro is not composable at all :(

Comment (tl-rodrigo-gryzinski) [reply reqwest-tracing/src/reqwest_otel_span_builder.rs:119]: 😢

**Review by conradludgate — APPROVED** [review]

**Pull request merged**